packages/video_player_avplayer/tizen/src/plus_player.cc

```diff
@@ -0,0 +1,666 @@
+  int64_t duration = 0;
+  if (player_->GetState() >= plusplayer::State::kTrackSourceReady) {
+    plusplayer::PlayerMemento memento;
+    if (!player_->GetMemento(&memento)) {
+      LOG_ERROR("[PlusPlayer] Fail to get player memento.");
+    }
+
+    if (memento.is_live) {
+      std::string str = player_->GetStreamingProperty("GET_LIVE_DURATION");
+      if (str.empty()) {
+        LOG_ERROR("[PlusPlayer] Fail to get live duration.");
+        return duration;
+      }
+      std::vector<std::string> time_str = split(str, '|');
+      int64_t start_time = std::stoll(time_str[0].c_str());
+      int64_t end_time = std::stoll(time_str[1].c_str());
+
+      duration = end_time - start_time;
+    } else {
+      if (!player_->GetDuration(&duration)) {
+        LOG_ERROR("[PlusPlayer] Fail to get the duration.");
+      }
+    }
+  }
+
+  LOG_INFO("[PlusPlayer] Video duration: %lld.", duration);
+  return duration;
+}
+
+void PlusPlayer::GetVideoSize(int32_t *width, int32_t *height) {
+  if (player_->GetState() >= plusplayer::State::kTrackSourceReady) {
+    bool found = false;
+    std::vector<plusplayer::Track> tracks = player_->GetActiveTrackInfo();
+    for (auto track : tracks) {
+      if (track.type == plusplayer::TrackType::kTrackTypeVideo) {
+        *width = track.width;
+        *height = track.height;
+        found = true;
+      }
+    }
+    if (!found) {
+      LOG_ERROR("[PlusPlayer] Failed to get video size.");
+    } else {
+      LOG_INFO("[PlusPlayer] Video widht: %d, height: %d.", *width, *height);
+    }
+  }
+}
+
+bool PlusPlayer::isReady() {
+  return plusplayer::State::kReady == player_->GetState();
+}
+
+bool PlusPlayer::SetDisplay() {
+  int x = 0, y = 0, width = 0, height = 0;
+  ecore_wl2_window_proxy_->ecore_wl2_window_geometry_get(native_window_, &x, &y,
+                                                         &width, &height);
+  int surface_id =
+      ecore_wl2_window_proxy_->ecore_wl2_window_surface_id_get(native_window_);
+  if (surface_id < 0) {
+    LOG_ERROR("[PlusPlayer] Failed to get surface id.");
+    return false;
+  }
+  bool ret = player_->SetDisplay(plusplayer::DisplayType::kOverlay, surface_id,
+                                 x, y, width, height);
+  if (!ret) {
+    LOG_ERROR("[PlusPlayer] Failed to set display.");
+    return false;
+  }
+
+  ret = player_->SetDisplayMode(plusplayer::DisplayMode::kDstRoi);
+  if (!ret) {
+    LOG_ERROR("[PlusPlayer] Failed to set display mode.");
+    return false;
+  }
+
+  return true;
+}
+
+flutter::EncodableList PlusPlayer::getTrackInfo(int32_t track_type) {
+  if (!player_) {
+    LOG_ERROR("[PlusPlayer] Player not created.");
+    return {};
+  }
+
+  plusplayer::State state = player_->GetState();
+  if (state == plusplayer::State::kNone || state == plusplayer::State::kIdle) {
+    LOG_ERROR("[PlusPlayer] Player is in invalid state.");
+    return {};
+  }
+
+  plusplayer::TrackType type;
+  switch (track_type) {
+    case 1:
```

Comment:
The track type could simply be a string.

```patch
@@ -104,9 +104,7 @@ class VideoPlayerTizen extends VideoPlayerPlatform {
   Future<List<VideoTrack>> getVideoTracks(int playerId) async {
     final TrackMessage response = await _api.track(TrackTypeMessage(
       playerId: playerId,
-      trackType: _intTrackTypeMap.keys.firstWhere(
-          (int key) => _intTrackTypeMap[key] == TrackType.video,
-          orElse: () => -1),
+      trackType: TrackType.video.name,
     ));

     final List<VideoTrack> videoTracks = <VideoTrack>[];
```

Then

```cpp
if (track_type == "video") {
  ...
```

Reply:
Fixed.
[Refactor track type](https://github.com/xiaowei-guan/plugins/pull/6/commits/1440c1a7d8c7e8d9e2c7c3fbfcfad545dd4d2d76)